dagger-core/src/main/java/io/odpf/dagger/core/source/parquet/splitassigner/ChronologyOrderedSplitAssigner.java

```diff
@@ -16,19 +20,39 @@
 import java.util.concurrent.PriorityBlockingQueue;
 import java.util.stream.Collectors;
 
+import static com.google.api.client.util.Preconditions.checkArgument;
+import static io.odpf.dagger.core.metrics.aspects.ChronologyOrderedSplitAssignerAspects.SPLITS_AWAITING_ASSIGNMENT;
+import static io.odpf.dagger.core.metrics.aspects.ChronologyOrderedSplitAssignerAspects.TOTAL_SPLITS_DISCOVERED;
+import static io.odpf.dagger.core.metrics.aspects.ChronologyOrderedSplitAssignerAspects.TOTAL_SPLITS_RECORDED;
+import static io.odpf.dagger.core.metrics.reporters.statsd.tags.ComponentTags.getSplitAssignerTags;
+
 public class ChronologyOrderedSplitAssigner implements FileSplitAssigner {
     private final PriorityBlockingQueue<InstantEnrichedSplit> unassignedSplits;
     private static final int INITIAL_DEFAULT_CAPACITY = 11;
     private PathParser pathParser;
     private TimeRangePool timeRangePool;
+    private DaggerGaugeManager daggerGaugeManager;
+    private final StatsDErrorReporter statsDErrorReporter;
 
-    private ChronologyOrderedSplitAssigner(Collection<FileSourceSplit> fileSourceSplits, PathParser pathParser, TimeRangePool timeRangePool) {
+    private ChronologyOrderedSplitAssigner(Collection<FileSourceSplit> fileSourceSplits, PathParser pathParser,
+                                           TimeRangePool timeRangePool, SerializedStatsDReporterSupplier statsDReporterSupplier) {
+        this.registerTagsWithMeasurementManagers(statsDReporterSupplier);
+        daggerGaugeManager.markValue(TOTAL_SPLITS_DISCOVERED, fileSourceSplits.size());
```

Comment:
Some of times we are using `this` and some of times we are not. It's better to be consistent.

Reply:
Will do

Reply:
Fixed via commit https://github.com/odpf/dagger/pull/174/commits/87ed606524e937e3183d6a9c0ba445eb19d610e2